lib/matplotlib/axes/_axes.py

```diff
@@ -6637,7 +6637,9 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
             bin_range = self.convert_xunits(bin_range)
 
         # Check whether bins or range are given explicitly.
-        binsgiven = np.iterable(bins) or bin_range is not None
+        binsgiven = ((np.iterable(bins) and
+                      not isinstance(bins, str)) or
+                     bin_range is not None)
```

Comment:
It might be worth matching the numpy logic here, as:
```python
if bins is None:
    binsgiven = False
else:
    # match the logic in numpy.lib.histograms._get_bin_edges
    if isinstance(bins, basestring):
        binsgiven = False
    elif np.ndim(bins) == 1:
        binsgiven = True
    else:
        binsgiven = False
```
Renaming to `bins_array_given` would make sense too.